File: Backend_Folders/src/main/java/org/example/backend_folders/controllers/FavoriteController.java
package org.example.backend_folders.controllers;

import jakarta.transaction.Transactional;
import org.example.backend_folders.remainderEntities.Favorite;
import org.example.backend_folders.services.FavoriteService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@CrossOrigin(origins = "http://localhost:4200")
@RestController
@RequestMapping("/api/favorite")
public class FavoriteController {

    private final FavoriteService favoriteService;

    public FavoriteController(FavoriteService favoriteService) {
        this.favoriteService = favoriteService;
    }

    @GetMapping
    public ResponseEntity<List<Favorite>> getFavorites(@RequestParam(value = "email", required = false) String email) {
        List<Favorite> favorites_list = favoriteService.getAllFavoritesByEmail(email);
        return new ResponseEntity<>(favorites_list, HttpStatus.OK);
    }

    @GetMapping("/exists")
    public ResponseEntity<Boolean> checkIfFavoriteExists(
            @RequestParam String email,
            @RequestParam String name) {
        Favorite element = favoriteService.findFavoriteWithNameAndEmail(email, name);
        boolean bool_element = element != null;
        return new ResponseEntity<>(bool_element, HttpStatus.OK);
    }


    @PostMapping
    public Favorite createFavorite(@RequestBody Favorite favorite) {
        return favoriteService.saveFavorite(favorite);
    }

    @Transactional
    @DeleteMapping("/deletion")
    public void deleteFavorite(@RequestParam String email,
                               @RequestParam String name) {
        favoriteService.deleteFavorite(email, name);
    }


}

File: backend-nest/src/favorite/favorite.controller.ts
import { Controller, Get, Post, Body, Patch, Param, Delete, Query, ValidationPipe, Logger } from '@nestjs/common';
import { FavoriteService } from './favorite.service';
import { CreateFavoriteDto } from './dto/create-favorite.dto';
import { UpdateFavoriteDto } from './dto/update-favorite.dto';

@Controller('/api/favorite')
export class FavoriteController {
  logger = new Logger(FavoriteService.name);

  constructor(private readonly favoriteService: FavoriteService) {}

  @Post()
  createFavorite(@Body(ValidationPipe) createFavoriteDto: CreateFavoriteDto) {
    return this.favoriteService.createFavorite(createFavoriteDto);
  }

  @Get()
  findFavorites(@Query('email') email: string) {
    return this.favoriteService.findAllFavoritesByEmail(email);
  }

  @Get('/exists')
  async verifyIfFavoriteExists(@Query('email') email: string, @Query('name') name: string) {
    const element = await this.favoriteService.findFavoriteWithEmailAndName(email, name);
    return element != null;
  }

  @Patch(':id')
  update(@Param('id') id: string, @Body(ValidationPipe) updateFavoriteDto: UpdateFavoriteDto) {
    return this.favoriteService.update(id, updateFavoriteDto);
  }

  @Delete('/deletion')
  removeFavorite(@Query('email') email: string, @Query('name') name: string) {
    return this.favoriteService.remove(email, name);
  }
}

/* @GetMapping("/exists")
    public ResponseEntity<Boolean> checkIfFavoriteExists(
            @RequestParam String email,
            @RequestParam String name) {
        Favorite element = favoriteService.findFavoriteWithNameAndEmail(email, name);
        boolean bool_element = element != null;
        return new ResponseEntity<>(bool_element, HttpStatus.OK);
    }


    @PostMapping
    public Favorite createFavorite(@RequestBody Favorite favorite) {
        return favoriteService.saveFavorite(favorite);
    }

 */